modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/formatFieldValue.es.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import moment from 'moment/min/moment-with-locales';
+
+export default (
+	{dataType, symbols, value},
+	preserveValue,
+	nextEditingLanguageId,
+	prevEditingLanguageId
+) => {
+	if (dataType === 'double') {
+		return String(value).replace(symbols.decimalSymbol, '.');
+	}
+
+	if (
+		dataType === 'date' &&
+		value.indexOf('_') === -1 &&
+		value !== '' &&
+		nextEditingLanguageId !== prevEditingLanguageId &&
+		preserveValue
+	) {
+		moment.locale(prevEditingLanguageId);
```

Comment:
@marcosapmf , is possible to use ```moment.locale(prevEditingLanguageId)``` directly using: ```moment.locale(prevEditingLanguageId).localeData().longDateFormat('L') ```

Or, create a variable (idk if returns anything) like: ```const momentWithPrevEditingLanguage = moment.locale(prevEditingLanguageId)``` ?

Why: In most cases, is not recommended to change value from instance directly, it may inject some moment configuration global, or spread someway to other moment instances.

@interaminense, have anything in mind ?

Reply:
I agree with you @kevenleone!

Reply:
Hey @kevenleone and @interaminense,
thanks a lot for reviewing this pull. As we've talked on slack I made this change and additionally pull out the date parse/format logic to another file. Could you please take a look at this again?

Thanks :D

Reply:
Sure